src/Breadcrumbs.php

```diff
@@ -147,77 +170,159 @@ public function links(array $value): self
+            ->attributes($listAttributes)
+            ->content("\n", $items, "\n")
+            ->id($listId)
+            ->encode(false)
+            ->render();
+    }
+
     /**
      * Renders a single breadcrumb item.
      *
-     * @param array $link the link to be rendered. It must contain the "label" element. The "url" element is optional.
-     * @param string $template the template to be used to rendered the link. The token "{link}" will be replaced by the
-     * link.
-     *
-     * @throws JsonException|RuntimeException if `$link` does not have "label" element.
+     * @param BreadcrumbLink $breadcrumbLink The breadcrumb item to render.
      *
-     * @return string the rendering result
+     * @return string The rendering result.
      */
-    private function renderItem(array $link, string $template): string
+    private function renderItem(BreadcrumbLink $breadcrumbLink): string
     {
-        $encodeLabel = ArrayHelper::remove($link, 'encode', $this->encodeLabels);
-
-        if (array_key_exists('label', $link)) {
-            $label = $encodeLabel ? Html::encode($link['label']) : $link['label'];
-        } else {
+        if ($breadcrumbLink->label === '') {
             throw new RuntimeException('The "label" element is required for each link.');
         }
 
-        if (isset($link['template'])) {
-            $template = $link['template'];
-        }
+        $itemsAttributes = $this->itemAttributes;
+        $classes = $itemsAttributes['class'] ?? null;
+
+        unset($itemsAttributes['class']);
+
+        $link = $this->renderLink($breadcrumbLink);
+        Html::addCssClass($itemsAttributes, [self::ITEM_NAME, $classes]);
 
-        if (isset($link['url'])) {
-            $options = $link;
-            unset($options['template'], $options['label'], $options['url']);
-            $linkHtml = Html::a($label, $link['url'], $options)->encode($this->encodeTags);
-        } else {
-            $linkHtml = $label;
+        if ($breadcrumbLink->url === null) {
+            $itemsAttributes['aria-current'] = 'page';
+
+            Html::addCssClass($itemsAttributes, $this->itemActiveClass);
         }
 
-        return strtr($template, ['{link}' => $linkHtml]);
+        return Li::tag()->attributes($itemsAttributes)->content($link)->encode(false)->render();
+    }
+
+    /**
+     * Renders a single breadcrumb link.
+     *
+     * @param BreadcrumbLink $breadcrumbLink The breadcrumb link to render.
+     *
+     * @return string The rendering result.
+     */
+    private function renderLink($breadcrumbLink): string
+    {
+        $label = Html::encode($breadcrumbLink->label);
+
+        return match ($breadcrumbLink->url) {
+            null => $label,
+            default => A::tag()
+                ->attributes($this->linkAttributes)
+                ->addAttributes($breadcrumbLink->getAttributes())
+                ->content($label)
+                ->url($breadcrumbLink->url)
+                ->encode(false)
```

Comment:
Looks a bit dangerous. Worth encoding.

Reply:
https://github.com/yiisoft/yii-bootstrap5/blob/a92b95bf55772e076ccf061b99db40327f1414c5/src/Breadcrumbs.php#L315 Label is always encoded.